technical-reference/reference-technical-articles/prime.md

```diff
@@ -0,0 +1,206 @@
+
+As the market won't be paused, there could be inconsistencies because there will be user supply/borrow transactions in between updating scores transactions. These inconsistencies will be very minor compared to letting it update gradually when users will borrow/supply.
+
+There are two main objectives for creating this script:
+
+- If the Venus community wants to update the scores of all users when multipliers or alpha are updated then we have an immediate option.
+- After minting Prime tokens if the Venus community decides to add an existing market to the Prime token program then the score of all users has to be updated to start giving them rewards. The scores cannot be applied gradually in this case as the initial Prime users for the market will get large rewards for some time. So this script will prevent this scenario.
+
+There is a variable named `totalScoreUpdatesRequired` to track how many scores updates are pending. This is for tracking purposes and visibility to the community.
+
+## Calculate APR associated with a Prime market and user
+
+The goal is to offer a view function that allows the [Venus UI](https://app.venus.io) to show an estimation of the APR associated with the Prime token and the borrow/supply position of a user.
+
+The steps to perform this calculation are:
+
+1. Fetch the income per block from PrimeLiquidityProvider
+2. Calculate the capped supply and borrow of the user for the market
+3. Calculate the capped XVS balance of the user 
+4. Calculate the user yearly income by multiplying (1) with blocks per year
+5. Now borrow and supply APR of user can be calculated based on ratio of capped borrow and supply of the user. 
+
+**Example:**
+
+1. Income per block 0.00003 USDT
+2. Income per year is 10512000 blocks/year * 0.00003 = 315.36 USDT
+3. Assuming the user score for USDT: 3, and the sum of scores for USDT: 10, then we would have 94.608 USDT (yearly income for this user, generated by Prime)
+4. Assuming the user has the following positions:
+   1. borrow: 30 USDT. Let's say it's capped at 15 USDT, so we'll consider 15 USDT
+   2. supply: 10 USDT. Let's say it's also capped at 15 USDT, so we'll consider 10 USDT
+5. Allocating the rewards (94.608 USDT), considering these capped versions, we would have:
+   1. borrow: 94.608 \* 15/25 = 56.76 USDT
+   2. supply: 94.608 \* 10/25 = 37.84 USDT
+6. Calculating the APR with these allocations, we would have:
+   1. borrow: 56.76/30 = 1.89 = 189%
+   2. supply: 37.84/10 = 3.78 = 378%
+
+Only part of the supplied and borrowed amounts (the capped amounts) are actually "working" on increasing the Prime rewards. The rest of the supplied or borrowed amounts do not generate extra rewards. In the example, if the user supplies more USDT, they won't generate more rewards (because the supply amount to be considered is capped at 15 USDT). So, it would make sense that the supply APR would decrease if they supply more USDT.
```

Comment:
```suggestion
Only the supply and borrow amounts below the cap generate Prime rewards. The supply and borrow amounts above the cap do not generate extra rewards. In the example, if the user supplies more USDT, they won't generate more rewards (because the supply amount to be considered is capped at 15 USDT). Therefore the supply APR would decrease if they supply more USDT.
```